docs/source/developer/node-with-devstack.rst

```diff
@@ -0,0 +1,274 @@
+Deploying Node with DevStack and Ironic
+=======================================
+
+This covers the steps to deploy a node using Ironic and set up DevStack with Ironic integration, including creating fake nodes in Ironic.
+
+.. contents:: Table of Contents
+   :depth: 2
+   :local:
+
+Image That We Used
+------------------
+
+For this, we used the Ubuntu 22.04 image to provision a node. You can use any image that is compatible with DevStack. You can download the image from the official Ubuntu website:
+
+[Download Ubuntu 22.04](https://releases.ubuntu.com/22.04/ubuntu-22.04-live-server-amd64.iso)
+
+Deploy the Node with an Image and Attach an External Floating IP
+----------------------------------------------------------------
+
+**Deploy the Node with the Image:**
+
+.. prompt:: bash $
+
+   metalsmith deploy --resource-class baremetal --image <image> --network <network> --candidate <node_id> --ssh-public-key <path-to-key>
+
+**Attach an External Floating IP:**
+
+1. Create a Floating IP:
+
+   .. prompt:: bash $
+
+     openstack floating ip create external
+
+2. List Floating IPs:
+
+   List all Floating IPs to see their details:
+
+   .. prompt:: bash $
+
+     openstack floating ip list
+
+3. List Ports:
+
+   Find the port ID of your node:
+
+   .. prompt:: bash $
+
+     openstack port list --network <network>
+
+4. Associate the Floating IP with the Port:
+
+   .. prompt:: bash $
+
+     openstack floating ip set --port <port_id> <floating_ip_id>
+
+**SSH into the Node Using the Floating IP:**
+
+Once the floating IP is attached, you can SSH into your node using the floating IP address:
+
+.. prompt:: bash $
+
+   ssh -i <path-to-key> ubuntu@<floating_ip>
+
+Prepare the Environment
+-----------------------
+
+**Update and upgrade the system packages:**
+
+.. prompt:: bash $
+
+   sudo apt update
+   sudo apt upgrade -y
+   sudo apt install -y git vim sudo curl
+
+**Verify Network Connectivity:**
+
+First, check if your instance has an active network connection by trying to ping a public IP address:
+
+.. prompt:: bash $
+
+   ping 8.8.8.8
+
+**Check DNS Configuration:**
+
+Ensure that your `/etc/resolv.conf` file is correctly configured with valid DNS servers. You can use Google's public DNS servers as a fallback:
+
+.. prompt:: bash $
+
+   sudo nano /etc/resolv.conf
+
+Add the following lines if they are not present:
+
+.. code-block:: text
+
+   nameserver 8.8.8.8
+   nameserver 8.8.4.4
+
+**Restart Network Services:**
+
+Restart the network manager to apply any changes made to the DNS configuration:
+
+.. prompt:: bash $
+
+   sudo systemctl restart network-manager
+
+**Update the system:**
+
+.. prompt:: bash $
+
+   sudo yum update -y
+   sudo yum upgrade -y
+
+**Install Git:**
+
+.. prompt:: bash $
+
+   sudo yum install git -y
+
+Install DevStack and configure it to work with Ironic by following these steps:
```

Comment:
as a title of a section, I'd reduce this to "Install DevStack and Configure Ironic"